From 83050ac67d40a449bc5433e0edda2c41d75eb45f Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Thu, 10 Jul 2014 23:14:49 +1000
Subject: [PATCH 1/6] Limit bitrate to 100Mbit/s and fill_min to half the size
 of the ringbuffer

Under some circumstances, FFmpeg can returns rubbish bitrates, set a maximum value of 100Mbit. This is almost twice the maximum bitrate permissible on a bluray.

Fixes #12196
---
 mythtv/libs/libmythtv/ringbuffer.cpp |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/mythtv/libs/libmythtv/ringbuffer.cpp b/mythtv/libs/libmythtv/ringbuffer.cpp
index 3ed6e5d..59c486a 100644
--- a/mythtv/libs/libmythtv/ringbuffer.cpp
+++ b/mythtv/libs/libmythtv/ringbuffer.cpp
@@ -324,6 +324,12 @@ void RingBuffer::UpdateRawBitrate(uint raw_bitrate)
             QString("Bitrate too low - setting to 64Kb"));
         raw_bitrate = 64;
     }
+    else if (raw_bitrate > 100000)
+    {
+        LOG(VB_FILE, LOG_INFO, LOC +
+            QString("Bitrate too high - setting to 100Mb"));
+        raw_bitrate = 100000;
+    }
 
     rwlock.lockForWrite();
     rawbitrate = raw_bitrate;
@@ -425,6 +431,7 @@ void RingBuffer::CalcReadAheadThresh(void)
         }
         low_buffers = false;
         fill_min = ((fill_min / CHUNK) + 1) * CHUNK;
+        fill_min = min((uint)fill_min, bufferSize / 2);
     }
     else
     {
-- 
1.7.10.2

